src/side-navigation/nav-item.js

```diff
@@ -56,3 +56,28 @@ export default class NavItem extends React.Component<NavItemPropsT> {
     );
   }
 }
+
+function areEqualShallow(a, b) {
+  if (!a || !b) return false;
+  if (typeof a !== 'object' || typeof a !== 'object') return false;
+
+  if (Object.keys(a).length !== Object.keys(b).length) return false;
+
+  for (var key in a) {
+    if (a[key] !== b[key]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function compare(prevProps, nextProps) {
```

Comment:
I've considered adding a new `memoComparison` prop to give consumers the ability to bail out of default comparison. What do others think about that approach?

Reply:
The optimization should be probably opt-in? It's not important for most of applications and brings potential bugs?

Reply:
👍that's a good idea